File: backend/src/modules/economy/entities/gift.entity.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
  Check,
} from 'typeorm';

@Entity('gifts')
@Check(`"coin_price" > 0`)
@Check(`"diamond_reward" > 0`)
export class Gift {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column({ type: 'varchar', length: 64 })
  name: string;

  @Column({ type: 'text' })
  icon_url: string;

  @Column({ type: 'text', nullable: true })
  animation_url: string | null;

  @Column({ type: 'bigint' })
  coin_price: string;

  @Column({ type: 'bigint' })
  diamond_reward: string;

  @Column({ type: 'boolean', default: true })
  is_active: boolean;

  @Column({ type: 'int', default: 0 })
  position: number;

  @CreateDateColumn({ type: 'timestamp with time zone' })
  created_at: Date;
}

File: android/app/src/main/java/com/chatqaq/app/data/local/entity/GiftEntity.kt
package com.chatqaq.app.data.local.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "gifts")
data class GiftEntity(
    @PrimaryKey
    val id: String,
    val name: String,
    @ColumnInfo(name = "icon_url")
    val iconUrl: String,
    @ColumnInfo(name = "animation_url")
    val animationUrl: String?,
    @ColumnInfo(name = "coin_price")
    val coinPrice: Long,
    @ColumnInfo(name = "diamond_reward")
    val diamondReward: Long,
    @ColumnInfo(name = "position", defaultValue = "0")
    val position: Int = 0
)
